src/woo_publications/api/tests/mixins.py

```diff
@@ -0,0 +1,181 @@
+from rest_framework import status
+from rest_framework.test import APIClient, APITestCase
+
+from .factories import TokenAuthFactory
+
+AUDIT_HEADERS = {
+    "AUDIT_USER_REPRESENTATION": "username",
+    "AUDIT_USER_ID": "id",
+    "AUDIT_REMARKS": "remark",
+}
+
+
+class APIKeyUnAuthorizedMixin:
+
+    self: APITestCase
+    client: APIClient
+
+    def assertWrongApiKeyProhibitsGetEndpointAccess(self, url):
+        no_permission_token = TokenAuthFactory.create().token
+        write_token = TokenAuthFactory.create(write_permission=True).token
+
+        with self.subTest("no token given"):  # type: ignore reportAttributeAccessIssue
+            response = self.client.get(url, headers=AUDIT_HEADERS)
+            assert response.status_code == status.HTTP_401_UNAUTHORIZED
```

Comment:
bare asserts is pytest-style, but we're doing unittest style, so this really needs to be

```suggestion
            self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)  # type: ignore reportAttributeAccessIssue
```
there's a reason these "wrappers" exist - it's so they can provide extra context and print appropriate messages to signal test failure.

Reply:
I did it this way to reduce the amount of `type: ignore reportAttributeAccessIssue` that was used in the file. But changed it based on your feedback.